.teamcity/Gradle_Check/configurations/FunctionalTest.kt

```diff
@@ -70,6 +68,10 @@ class FunctionalTest(
                 param("env.ANDROID_HOME", """C:\Program Files\android\sdk""")
             }
         }
+
+        if(testCoverage.testDistribution) {
+            param("maxParallelForks", "16")
```

Comment:
I think it would be clearer if this logic would be part of the build in `Project.maxParallelForks` in `unittest-and-compile.gradle.kts`. We can check `enableTestDistribution` there. Wdyt?

Reply:
We had this value set to 4 before:

```
        "-Dorg.gradle.workers.max=%maxParallelForks%",
        "-PmaxParallelForks=%maxParallelForks%",
```

This configuration means 4 parallel test tasks, and each task has 4 parallel forks, which means 4x4=16 test classes running in parallel, right?

My intention was "16 parallel tasks", this seems like a global property which can only be configured via `-Dorg.gradle.workers.max`, right? Only "how many test classes can be executed in parallel" can be configured in `unittest-and-compile.gradle.kts`?

Reply:
`org.gradle.workers.max` is the upper limit for the parallel forks across all test tasks. So if it is 4, then there will be at maximum 4 test classes running in parallel. If you want to have at most 16 test classes running in parallel, then this needs to be 16. `maxParallelForks` only controls the parallelism per test task, though it can never surpass the global max workers count.

Reply:
I see, so every test class occupies a worker slot. Then I want 16 test tasks running in parallel (there's no test classes running because it was handled by test distribution), I have to configure `org.gradle.workers.max=16` in command line, right?

Reply:
If test distribution does not respect the worker count for test classes running remote executors (which makes sense), then `org.gradle.workers.max=16` will cause at most 16 test tasks to run in parallel. Gradle's built-in test executor will acquire a worker lease per test JVM fork.

> Then I want 16 test tasks running in parallel (there's no test classes running because it was handled by test distribution), I have to configure org.gradle.workers.max=16 in command line, right?

I suppose that should work.